eureca_building/systems.py

```diff
@@ -136,34 +160,56 @@ def set_dhw_design_capacity_tank(self, dhw_flow_rate, weather_file):
                                water_properties["density"] * \
                                water_properties["specific_heat"] * \
                                self.dhw_tank_design_delta_T / 3600    # [Wh]
+        self.dhw_tank_maximum_charge=self.dhw_tank_volume * \
+                               water_properties["density"] * \
+                               water_properties["specific_heat"] * \
+                               (T_tank_max-T_user) / 3600          
+        self.dhw_tank_minimum_charge=self.dhw_tank_volume * \
+                               water_properties["density"] * \
+                               water_properties["specific_heat"] * \
+                               (T_tank_min-T_user) / 3600          
         self.dhw_tank_design_charge  = 1e-10 if self.dhw_tank_design_charge  < 1e-10 else self.dhw_tank_design_charge
 
         self.dhw_tank_current_charge = self.dhw_tank_design_charge / 2
         self.dhw_tank_current_charge_perc = 0.5
         self.charging_mode = 0.
+        self.discharging_mode =0.
         self.dhw_capacity_to_tank = 0.
+        self.dhw_tank_current_charge_perc=self.dhw_tank_minimum_charge/self.dhw_tank_design_charge
         self.losses_discharging_rate = 0.02 # %/h
 
-    def dhw_tank_solver(self, dhw_demand):
-
-        # if ST exists:
-        #     ST_prodction = ST prod[t]
-        # else
-        #     ST_prodction = 0 # Wh
-
-        if self.dhw_tank_current_charge < 0:
-            self.charging_mode = 1 # Turn on  system
-            self.dhw_capacity_to_tank = self.dhw_design_load
-        elif self.dhw_tank_current_charge > self.dhw_tank_design_charge:
-            self.charging_mode = 0 # Turn off  system
-            self.dhw_capacity_to_tank = 0.
+    def dhw_tank_solver(self, dhw_demand, weather,timestep,**kwargs):
+        if hasattr(self,"solar_gain"):
+            self.solar_thermal_gain=self.solar_gain[timestep]/CONFIG.ts_per_hour
+        else:
+            self.solar_thermal_gain=0
```

Comment:
Eviterei di creare un altro attributo self.solar_thermal_gain quando c'è gia self.solar_gain

metterei così:

solar_gain = self.solar_gain[timestep]/CONFIG.ts_per_hour if isinstance("solar_gain", np.array) else 0

In questo modo vado a prendere il numpy se c'è st altrimento prende 0

Forse si possono togliere tutti i solar_gain = 0 in tutte le classi. Ne parliamo